src/assets/scripts/client/airportGuide/AirportGuideViewController.js

```diff
@@ -0,0 +1,219 @@
+
+        /**
+         * Contains an object with keys of icao idents and values
+         * of the airport guides from the documentation folder.
+         *
+         * @property _guideData
+         * @type {object<string, string>}
+         * @private
+         */
+        this._guideData = airportGuideData;
+
+        /**
+         * Local reference to the EventBus
+         *
+         * @property _eventBus
+         * @type {EventBus}
+         * @private
+         */
+        this._eventBus = null;
+
+        return this._init()
+            ._setupHandlers()
+            .enable();
+    }
+
+    /**
+     * Initialize the instance
+     *
+     * Should only be run once on instantiation
+     *
+     * @for airportGuideViewController
+     * @method init
+     * @chainable
+     */
+    _init() {
+        this._eventBus = EventBus;
+
+        return this;
+    }
+
+    /**
+     * Bind method handlers
+     *
+     * Should only be run once on instantiation
+     *
+     * @for airportGuideViewController
+     * @method _setupHandlers
+     * @chainable
+     */
+    _setupHandlers() {
+        this._onAirportChangeHandler = this._onAirportChange.bind(this);
+        this._onElementToggleHandler = this._onToggleAirportGuide.bind(this);
+
+        return this;
+    }
+
+    /**
+     * Enable handlers
+     *
+     * @for airportGuideViewController
+     * @method enable
+     * @chainable
+     */
+    enable() {
+        this._eventBus.on(EVENT.AIRPORT_CHANGE, this._onAirportChangeHandler);
+        this.$airportGuideTrigger.on('click', this._onElementToggleHandler);
+
+        this.airportGuideViewModel = new AirportGuideViewModel(this.initialIcao, this.getAirportGuide(this.initialIcao));
+
+        this.$element.append(this.airportGuideViewModel.$element);
+
+        return this;
+    }
+
+    /**
+     * Disable handlers
+     *
+     * @for airportGuideViewController
+     * @method disable
+     * @chainable
+     */
+    disable() {
+        this._eventBus.off(EVENT.AIRPORT_CHANGE, this._onAirportChangeHandler);
+        this.$airportGuideTrigger.off('click', this._onElementToggleHandler);
+
+        return this;
+    }
+
+    /**
+     * Reset the instance
+     *
+     * @for airportGuideViewController
+     * @method destroy
+     * @chainable
+     */
+    destroy() {
+        this._eventBus = null;
+        this.$element = null;
+        this.initialIcao = null;
+        this.airportGuideViewModel = null;
+        this.$airportGuideTrigger = null;
+        this._guideData = null;
+        this._onAirportChangeHandler = null;
+        this._onElementToggleHandler = null;
+
+        return this;
+    }
+
+    /**
+     * Event handler for when a `airportGuideView` instance is clicked
+     *
+     * @for airportGuideViewController
+     * @method _onToggleAirportGuide
+     * @param event {JQueryEventObject}
+     * @private
+     */
+    _onToggleAirportGuide() {
+        this.airportGuideViewModel.toggleView();
```

Comment:
We should toggle the `active` class on the button in the footer

Reply:
correct.  the opening and closing of the modal should be handled by the `UiController` (next thing I'm tackling), and within that, we toggle the icon's active class